React/Views/RCTWrapperViewController.m

```diff
@@ -79,6 +79,11 @@ - (void)viewWillAppear:(BOOL)animated
       [bar setTitleTextAttributes:@{NSForegroundColorAttributeName : _navItem.titleTextColor}];
     }
 
+    if (_navItem.shadowHidden) {
```

Comment:
You should make sure to set hidden back to NO if this changes from true to false.